src/lib/types.ts

```diff
@@ -1,25 +1,31 @@
 import { z } from "zod";
+import zxcvbn from "zxcvbn";
 
 export const zPassword = z
   .string({
-    required_error: "Veuillez renseigner un mot de passe",
+    required_error: "Le mot de passe n'est pas assez fort",
   })
-  .min(6, {
-    message: "Le mot de passe doit contenir au moins 6 caractères",
-  })
-  .refine(
-    (value) => /^.*[A-Z].*$/.test(value),
-    "Le mot de passe doit contenir au moins une majuscule",
-  )
-  .refine(
-    (value) => /^.*[a-z].*$/.test(value),
-    "Le mot de passe doit contenir au moins une minucule",
-  )
-  .refine(
-    (value) => /^.*[0-9].*$/.test(value),
-    "Le mot de passe doit contenir au moins un chiffre",
-  )
-  .refine(
-    (value) => /^.*[!@#$%^&*(),.?":{}|<>\-_[\]+=;].*$/.test(value),
-    'Le mot de passe doit contenir au moins un caractère spécial parmi les suivants [!@#$%^&*(),.?":{}|<>]-_',
-  );
+  .superRefine((value, ctx) => {
+    const zxcvbnResult = zxcvbn(value || "");
+
+    if (zxcvbnResult.score >= 6) {
```

Comment:
Shouldn't it be 5, since zxcvbn returns a value between 0 and 4 and we add 1 to it for display purposes ?